src/geotiff.js

```diff
@@ -610,25 +610,26 @@ class MultiGeoTIFF extends GeoTIFFBase {
    * @returns {GeoTIFFImage} the image at the given index
    */
   async getImage(index = 0) {
-    if (!this.fileDirectoriesPerFile) {
-      if (!this.fileDirectoriesPerFileParsing) {
-        this.fileDirectoriesPerFileParsing = this.parseFileDirectoriesPerFile();
+    await this.getImageCount();
+    await this.parseFileDirectoriesPerFile();
+    let visited = 0;
+    let relativeIndex = 0;
+    for (let i = 0; i < this.imageFiles.length; i++) {
+      const imageFile = this.imageFiles[i];
+      for (let ii = 0; ii < this.imageCounts[i]; ii++) {
+        if (index === visited) {
+          const ifd = await imageFile.requestIFD(relativeIndex);
+          return new GeoTIFFImage(
+            ifd.fileDirectory, imageFile.geoKeyDirectory,
+            imageFile.dataView, imageFile.littleEndian, imageFile.cache, imageFile.source,
+          );
```

Comment:
I think there is something slight wrong with what I'm doing here @constantinius but i'm still trying to figure it out...

Reply:
I'm trying to do a 
````
await tiff.readRasters({
      bbox: imgBbox,
      width: 512,
      height: 512
    })
````
And I'm getting a `(node:47810) UnhandledPromiseRejectionWarning: buffer error`

Reply:
Actually nope I think that was a dodgy implementation thing I had somewhere.